frontend/src/containers/App.js

```diff
@@ -0,0 +1,194 @@
+        pc.createOffer(
+          function(localDescription) {
+            console.log('set location description for newPeerId=%o', newPeerId)
+            pc.setLocalDescription(localDescription,
+            function() {
+              // With the local description, we can send the event. We will
+              // await a 'ReceiveMediaAnswer' event and set the remote
+              // description on this peer connection.
+              let data = {
+                'from_peer_id': newPeerId,
+                'to_peer_id':   myPeerId,
+                'desc':         btoa(JSON.stringify(pc.localDescription))
+              }
+              rmsClient.receiveMediaFrom(data, () => {
+                console.log('peerId=%o requested to receive media from peerId=%o', myPeerId, newPeerId);
+                rmsClient.awaitReceiveMediaAnswer((resp) => {
+                  console.log('received media answer resp=%o for peerId=%o', resp, newPeerId);
+                  // TODO: validate.
+                  let sdpAnswer = JSON.parse(atob(resp["sdp_answer"]));
+                  if (sdpAnswer !== '') {
+                    var remoteDescription = new RTCSessionDescription(sdpAnswer);
+                    var tmp = pc.setRemoteDescription(remoteDescription,
+                    function() {
+                      console.log('set remote description for peerId=%o', newPeerId);
+                    }, function (e) {
+                      console.log('error=%o setting remote description for peerId=%o', e, newPeerId);
+                    });
+                    console.log('remote description=%o for peerId=%o', newPeerId)
+                  }
+                });
+              });
+            });
+          }
+        );
+      });
+    });
+  }
+
+  return (
+    <div className="App">
+      <div className="header">
+        <h1>Roomz Media Server Playground</h1>
+      </div>
+      <div className="vestibule-video-preview-ctnr">
+        <div className="video" id="vestibule-video-preview">
+          <label htmlFor="outgoing-vid">Vestibule Video</label><br/>
+          <video ref={egress_media} id="vestibule-vid" autoPlay controls/>
+        </div>
+        <div className="user-actions">
+          <button className="roomz-btn button-primary" onClick={setupLocalMedia}>Setup Media</button>
+        </div>
+      </div>
+      <div className="user-settings">
+        <form className="user-settings-form">
+          <div className="user-input-form">
+            <label htmlFor="room-id">Room Id: </label>
+            <input id="room-id" ref={room_id} autoFocus/>
+          </div>
+          <div className="user-input-form">
+            <label htmlFor="user-id">User Id: </label>
+            <input id="user-id" ref={user_id} autoFocus/>
+          </div>
+        </form>
+        <div className="user-actions">
+          <button className="roomz-btn button-primary" onClick={joinMediaRoom}>JoinMediaRoom</button>
+        </div>
+      </div>
+    </div>
```

Comment:
No change needed for this PR, but splitting this up can help make this more readable. Not sure if there is a standardized way to do this. I guess since the "split-up into multiple functions" is already done in the main frontend repo, it could be done here too.